src/segment-loader.js

```diff
@@ -757,12 +762,11 @@ export default class SegmentLoader extends videojs.EventTarget {
     let currentTime = this.currentTime_();
 
     this.pendingSegment_ = null;
-    // add segment timeline information if we're still using the
-    // same playlist
-    if (segmentInfo && segmentInfo.playlist.uri === this.playlist_.uri) {
-      this.updateTimeline_(segmentInfo);
-      this.trigger('progress');
-    }
+
+    // add segment metadata if it we have gained information during the
+    // last append
+    this.updateTimeline_(segmentInfo);
+    this.trigger('progress');
```

Comment:
Is it OK that we are now triggering progress even if we have switched playlists, whereas we weren't before?

For instance, if we go to appendBuffer, playlist is switched, and we just go ahead and append the buffer after appendBuffer finishes, we have progress for the old rendition, but not the current one. Should that count?